main.tf

```diff
@@ -255,29 +247,21 @@ resource "azurerm_monitor_diagnostic_setting" "acr-diag" {
   storage_account_id         = var.storage_account_id
   log_analytics_workspace_id = var.log_analytics_workspace_id
 
-  dynamic "log" {
-    for_each = var.acr_diag_logs
+  dynamic "enabled_log" {
+    for_each = var.log_enabled ? ["allLogs"] : []
```

Comment:
why directly we are passing `allLogs` can we not enable other types of logs?

Reply:
this is a better option to get detailed logs.

Reply:
we can set it by default, as some might want only audit logs or something else.